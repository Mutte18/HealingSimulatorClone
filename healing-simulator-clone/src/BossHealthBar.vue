<template>
  <div class="outerHealthBar" style="background-color: darkgoldenrod">
    <div class="innerHealthBar" :style="getHealthPointStyle"></div>
    <div class="healthBarText">
      {{ bossName }}
      <br>
      {{ getHealthAsPercentage }}
    </div>
  </div>
</template>

<script>
    export default {
      props: {
        healthPoints: {
          type: Number,
          required: true
        },
        maxHealth: {
          type: Number,
          required: true
        },
        bossName: {
          type: String,
          required: true
        }
      },
      computed: {
        getHealthAsPercentage(){
          let percentage = (this.healthPoints / this.maxHealth) * 100;
          return percentage + '%'
        },
        getHealthPointStyle(){
          let percentage = (this.healthPoints / this.maxHealth) * 100;
          if(percentage <= 0)
          {
            percentage = 0;
          }
          return {
            width: percentage + '%'
          }
        }
      }
    }
</script>

<style scoped>
  .outerHealthBar{
    width: 75%;
    height: 50px;
    margin-left: 15%;
  }

  .healthBarText{
    position: relative;
    text-align: center;
    color: white;
    bottom: 40px;
  }

  .innerHealthBar{
    width: inherit;
    position: relative;
    height: 50px;
    transition: width 500ms;
    background-color: darkgreen;
  }
</style>
