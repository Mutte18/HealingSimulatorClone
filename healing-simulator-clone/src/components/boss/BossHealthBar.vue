<template>
  <div class="outerHealthBar" style="background-color: darkgoldenrod">
    <div class="innerHealthBar" :style="getHealthPointStyle"></div>
    <div class="healthBarText">
      {{ bossName }}
      <br>
      {{ getHealthAsPercentage }}
    </div>
  </div>
</template>

<script>
export default {
  props: {
    healthPoints: {
      type: Number,
      required: true,
    },
    maxHealth: {
      type: Number,
      required: true,
    },
    bossName: {
      type: String,
      required: true,
    },
  },
  computed: {
    getHealthAsPercentage() {
      const percentage = (this.healthPoints / this.maxHealth) * 100;
      return `${percentage}%`;
    },
    getHealthPointStyle() {
      let percentage = (this.healthPoints / this.maxHealth) * 100;
      if (percentage <= 0) {
        percentage = 0;
      }
      return {
        width: `${percentage}%`,
      };
    },
  },
};
</script>

<style scoped>
  .outerHealthBar{
    width: 1000px;
    height: 50px;
    border: solid 1px black;

  }

  .healthBarText{
    position: relative;
    text-align: center;
    color: white;
    bottom: 40px;
  }

  .innerHealthBar{
    width: inherit;
    position: relative;
    height: 50px;
    transition: width 500ms;
    background-color: darkgreen;
  }
</style>
