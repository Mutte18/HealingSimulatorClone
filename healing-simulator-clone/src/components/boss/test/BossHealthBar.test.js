import { shallowMount } from "@vue/test-utils";
import BossHealthBar from '../BossHealthBar.vue';

const factory = (propsData) => {
  return shallowMount(BossHealthBar, {
    propsData: {
      maxHealth: 100,
      bossName: 'test',
      ...propsData
    }
  })
};



test('renders correctly', () => {
  const wrapper = factory({
    healthPoints: 100,
  });
  expect(wrapper).toMatchSnapshot();
});

test('getHealthPointStyle with 0 hp', () => {
  const wrapper = factory({
    healthPoints: 0,
  });
  expect(wrapper).toMatchSnapshot();
})
