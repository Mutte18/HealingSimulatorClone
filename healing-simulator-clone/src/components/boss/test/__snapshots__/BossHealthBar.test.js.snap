// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHealthPointStyle with 0 hp 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
